packages/virtualdom/src/index.ts

```diff
@@ -856,15 +904,18 @@ export function h(tag: string): VirtualElement {
       children.push(arg);
     } else if (arg instanceof VirtualElement) {
       children.push(arg);
-    } else if (arg instanceof VirtualElementPass) {
-      children.push(arg);
     } else if (arg instanceof Array) {
       extend(children, arg);
-    } else if (i === 1 && arg && typeof arg === 'object') {
-      attrs = arg;
+    } else if ((i === 1 || i === 2) && arg && typeof arg === 'object') {
+      if ("render" in arg && "unrender" in arg) {
```

Comment:
Since `unrender` is optional, we shouldn't check for it here, right?

Reply:
Yes, good catch. Removing the check for `unrender` will weaken the run time type constraint, but I just doubled checked and it looks like `render` is never used as an html attribute, so it should be fine

Reply:
Done. Removing `unrender` from the runtime typecheck made me a little bit nervous since it doesn't leave much to distinguish an `IRenderer` from an attributes object. It should be fine, though, since `render` is never a valid html attribute (I checked the spec)